### PR TITLE
Implements world admins addition functionality

### PR DESCRIPTION
**Pull request by mike-lvov**
closes https://github.com/sparkletown/internal-sparkle-issues/issues/1657

### CODE REVIEW

File: functions/src/utils/world.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import * as admin from "firebase-admin";
+import { HttpsError } from "firebase-functions/v1/https";
+
+type GetWorldBySlugProps = {
```
Comment (goran-peoski-work) [inline functions/src/utils/world.ts:4]:
```suggestion
type GetWorldBySlugOptions = {
```

File: functions/src/utils/world.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  worldSlug: string;
+};
+
+export const getWorldBySlug = async ({ worldSlug }: GetWorldBySlugProps) => {
```
Comment (goran-peoski-work) [inline functions/src/utils/world.ts:8]:
```suggestion
export const getWorldBySlug = async ({ worldSlug }: GetWorldBySlugOptions) => {
```

File: functions/src/utils/world.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  const matchingWorlds = await admin
+    .firestore()
+    .collection("worlds")
```
Comment (goran-peoski-work) [inline functions/src/utils/world.ts:11]:
```suggestion
    .collection(COLLECTIONS_WORLDS)
```

File: functions/src/utils/world.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    .where("slug", "==", worldSlug)
```
Comment (goran-peoski-work) [inline functions/src/utils/world.ts:12]:
```suggestion
    .where(FIELD_SLUG, "==", worldSlug)
```

File: functions/src/utils/world.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    .get();
+
+  if (matchingWorlds.empty) {
+    throw new HttpsError(
+      "internal",
+      `The world with ${worldSlug} does not exist`
+    );
+  }
+
+  const [world] = matchingWorlds.docs;
```
Comment (goran-peoski-work) [inline functions/src/utils/world.ts:22]: No check for `docs.length > 1` for Bugsnag?

File: functions/src/analytics.ts [hunk 6]
```diff
@@ -69,22 +70,14 @@ const generateAnalytics: HttpsFunctionHandler<{
     throw new HttpsError("internal", `No authentication context`);
   }
 
-  await throwErrorIfNotSuperAdmin(context.auth.token.user_id);
+  if (!data.worldSlug) {
+    throw new HttpsError("internal", `World slug was not passed`);
```
Comment (goran-peoski-work) [inline functions/src/analytics.ts:74]:
```suggestion
    throw new HttpsError("internal", `World slug missing`);
```

File: functions/src/user.ts [hunk 7]
```diff
@@ -0,0 +1,46 @@
+import * as admin from "firebase-admin";
+import * as functions from "firebase-functions";
+import { HttpsError } from "firebase-functions/v1/https";
+
+import { HttpsFunctionHandler } from "./types/utility";
+import { throwErrorIfNotWorldOwner } from "./utils/permissions";
+import { getWorldBySlug } from "./utils/world";
+
+export const makeUserWorldOwner: HttpsFunctionHandler<{
+  worldSlug: string | undefined;
```
Comment (goran-peoski-work) [inline functions/src/user.ts:10]: Branded maybe? Does it make sense?

```suggestion
  worldSlug?: WorldSlug | string;
```

File: src/pages/Admin/ToolsPage/scripts/index.ts [hunk 8]
```diff
@@ -1 +1,2 @@
 export * from "./analytics/config";
+export * from "./makeWorldOwner/config";
```
Comment (goran-peoski-work) [inline src/pages/Admin/ToolsPage/scripts/index.ts:2]: Shouldn't these be `./analytics` and `./makeWorldOwner`?
Is adding `index.ts` in them going to mess up anything?

File: src/pages/Admin/ToolsPage/scripts/makeWorldOwner/AddWorldOwnerOutput.tsx [hunk 9]
```diff
@@ -0,0 +1,15 @@
+import React from "react";
+
+export interface AddWorldOwnerOutputProps {
+  userEmail: string;
+}
+
+export const AddWorldOwnerOutput: React.FC<AddWorldOwnerOutputProps> = ({
+  userEmail,
+}) => {
+  return (
```
Comment (goran-peoski-work) [inline src/pages/Admin/ToolsPage/scripts/makeWorldOwner/AddWorldOwnerOutput.tsx:10]:
```suggestion
}) => (
```

**Review by goran-peoski-work — APPROVED** [review]
Mostly code cosmetics. Should be good to go as is.

File: functions/src/user.ts (hunk 7 above)
Comment (mike-lvov) [reply functions/src/user.ts:10]: We don't yet have `branded` in function. I wonder if it's worth adding this concept here given that we can't reuse it from our core codebase. @colinhowe wdyt?

File: functions/src/utils/world.ts (hunk 5 above)
Comment (mike-lvov) [reply functions/src/utils/world.ts:22]: It checks if it's empty 7 line above

File: src/pages/Admin/ToolsPage/scripts/makeWorldOwner/AddWorldOwnerOutput.tsx (hunk 9 above)
Comment (mike-lvov) [reply src/pages/Admin/ToolsPage/scripts/makeWorldOwner/AddWorldOwnerOutput.tsx:10]: Done

File: src/pages/Admin/ToolsPage/scripts/index.ts (hunk 8 above)
Comment (mike-lvov) [reply src/pages/Admin/ToolsPage/scripts/index.ts:2]: Done

File: functions/src/utils/world.ts (hunk 3 above)
Comment (mike-lvov) [reply functions/src/utils/world.ts:11]: https://github.com/sparkletown/sparkle/pull/2834/files#r805986281

File: functions/src/utils/world.ts (hunk 4 above)
Comment (mike-lvov) [reply functions/src/utils/world.ts:12]: https://github.com/sparkletown/sparkle/pull/2834/files#r805986281

**Pull request merged**